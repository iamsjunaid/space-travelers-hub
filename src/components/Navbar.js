import React, { useState, useEffect } from 'react';
import { Route, Routes, Link } from 'react-router-dom';
import logo from '../images/planet.png';
import '../styles/Navbar.css';
import RocketsList from './navigation/RocketsList';
import MissionsList from './navigation/MissionsList';
import DragonsList from './navigation/DragonsList';
import MyProfile from './navigation/MyProfile';

function Navbar() {
  const [activeLink, setActiveLink] = useState('/');

  useEffect(() => {
    setActiveLink('/rockets'); // Set '/rockets' as the default active link
  }, []);

  return (
    <>
      <nav>
        <ul>
          <li>
            <Link
              to="/rockets"
              onClick={() => setActiveLink('/rockets')}
              className={activeLink === '/rockets' ? 'active' : ''}
            >
              <img src={logo} alt="logo" />
            </Link>
          </li>
          <li>title</li>
        </ul>
        <ul>
          <li>
            <Link
              to="/rockets"
              onClick={() => setActiveLink('/rockets')}
              className={activeLink === '/rockets' ? 'active' : ''}
            >
              Rockets
            </Link>
          </li>
          <li>
            <Link
              to="/missions"
              onClick={() => setActiveLink('/missions')}
              className={activeLink === '/missions' ? 'active' : ''}
            >
              Missions
            </Link>
          </li>
          <li>
            <Link
              to="/dragons"
              onClick={() => setActiveLink('/dragons')}
              className={activeLink === '/dragons' ? 'active' : ''}
            >
              Dragons
            </Link>
          </li>
          <li>
            <Link
              to="/profile"
              onClick={() => setActiveLink('/quote')}
              className={activeLink === '/quote' ? 'active' : ''}
            >
              My Profile
            </Link>
          </li>
        </ul>
      </nav>

      <Routes>
        <Route path="/" />
        <Route path="/rockets" element={<RocketsList />} />
        <Route path="/missions" element={<MissionsList />} />
        <Route path="/dragons" element={<DragonsList />} />
        <Route path="/profile" element={<MyProfile />} />
      </Routes>
    </>
  );
}

export default Navbar;
