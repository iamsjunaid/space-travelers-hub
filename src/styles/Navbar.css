nav {
  display: flex;
  justify-content: space-between;
}

nav ul {
  display: flex;
}

nav li {
  list-style: none;
  padding: 1em;
}

nav li a {
  text-decoration: none;
  color: black;
}

nav img {
  width: 50px;
  height: 50px;
}

.active {
  text-decoration: underline;
}
