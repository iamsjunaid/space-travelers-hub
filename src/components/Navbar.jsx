import React, { useState } from 'react';
import { Route, Routes, Link } from 'react-router-dom';
import logo from '../images/planet.png';
import '../styles/Navbar.css';
import Rockets from './navigation/Rockets';
import Missions from './navigation/Missions';
import Dragons from './navigation/Dragons';
import MyProfile from './navigation/MyProfile';

function Navbar() {
  const [activeLink, setActiveLink] = useState('/'); // Set initial active link to "/"
  return (
    <>
      <nav>
        <ul>
          <li>
            <Link
              to="/"
              onClick={() => setActiveLink('/')}
              className={activeLink === '/' ? 'active' : ''}
            >
              <img src={logo} alt="logo" />
            </Link>
          </li>
          <li>title</li>
        </ul>
        <ul>
          <li>
            {' '}
            <Link
              to="/rockets"
              onClick={() => setActiveLink('/rockets')}
              className={activeLink === '/rockets' ? 'active' : ''}
            >
              Rockets
            </Link>
          </li>
          <li>
            <Link
              to="/missions"
              onClick={() => setActiveLink('/missions')}
              className={activeLink === '/missions' ? 'active' : ''}
            >
              Missions
            </Link>
          </li>
          <li>
            <Link
              to="/dragons"
              onClick={() => setActiveLink('/dragons')}
              className={activeLink === '/dragons' ? 'active' : ''}
            >
              Dragons
            </Link>
          </li>
          <li>
            <Link
              to="/profile"
              onClick={() => setActiveLink('/quote')}
              className={activeLink === '/quote' ? 'active' : ''}
            >
              My Profile
            </Link>
          </li>
        </ul>
      </nav>

      <Routes>
        <Route path="/" />
        <Route path="/rockets" element={<Rockets />} />
        <Route path="/missions" element={<Missions />} />
        <Route path="/dragons" element={<Dragons />} />
        <Route path="/profile" element={<MyProfile />} />
      </Routes>
    </>
  );
}

export default Navbar;
