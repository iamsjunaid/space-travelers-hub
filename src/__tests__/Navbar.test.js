import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Navbar from "../components/NavBar";

test('renders Navbar correctly', () => {
  const { container } = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>,
  );

  expect(container).toMatchSnapshot();
});
