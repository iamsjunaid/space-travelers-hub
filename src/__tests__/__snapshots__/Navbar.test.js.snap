// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar correctly 1`] = `
<div>
  <nav
    class="navbar"
  >
    <ul>
      <li>
        <a
          class="active"
          href="/"
        >
          <img
            alt="logo"
            src="planet.png"
          />
        </a>
      </li>
      <li>
        <h1>
          Space Travelers' Hub
        </h1>
      </li>
    </ul>
    <ul>
      <li>
        <a
          class="active"
          href="/rockets"
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          class=""
          href="/missions"
        >
          Missions
        </a>
      </li>
      <li>
        <a
          class=""
          href="/dragons"
        >
          Dragons
        </a>
      </li>
      <li
        class="vertical-bar"
      />
      <li>
        <a
          class=""
          href="/profile"
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
